gbpservice/nfp/lifecycle_manager/drivers/lifecycle_driver_base.py

```diff
@@ -0,0 +1,440 @@
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+from oslo_log import log as logging
+
+from gbpservice.nfp._i18n import _
+from gbpservice.nfp.lifecycle_manager.openstack import (
+    openstack_driver
+)
+
+LOG = logging.getLogger(__name__)
+
+
+class LifeCycleDriverBase(object):
+    """Generic Driver class for Lifecycle handling of virtual appliances
+
+    Does not support sharing of virtual appliance for different chains
+    Does not support hotplugging interface to devices
+    Launches the VM with all the management and data ports and a new VM
+    is launched for each Network Service Instance
+    """
+    def __init__(self, supports_device_sharing=False, supports_hotplug=False,
+                 max_interfaces=5):
+        self.service_vendor = 'general'
+        self.supports_device_sharing = supports_device_sharing
+        self.supports_hotplug = supports_hotplug
+        self.maximum_interfaces = max_interfaces
+
+        # TODO[Magesh]: Try to move the following handlers to
+        # Device LCM manager rather than having here in the driver
+        self.identity_handler = openstack_driver.KeystoneClient()
+        self.compute_handler_nova = openstack_driver.NovaClient()
+        self.network_handler_gbp = openstack_driver.GBPClient()
+        self.network_handler_neutron = openstack_driver.NeutronClient()
+
```

Comment:
Need to discuss:
1. Continuing prev comment, these should be:
identity_api = lifecycle_manager.auth.api.AuthAPI()
compute_api = lifecycle_manager.compute.api.AuthAPI()
network_api = lifecycle_manager.network.api.AuthAPI()
2. gbp and neutron can be handled by network_api, and the network.api will use a driver for gbp, neutron
3. If we use this all the api will move down one level to lifecycle_manager/infra/auth ... etc.